lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -537,12 +537,22 @@ def create_transaction_parameters(money, credit_card_or_vault_id, options)
             :first_name => credit_card_or_vault_id.first_name,
             :last_name => credit_card_or_vault_id.last_name
           )
-          parameters[:credit_card] = {
-            :number => credit_card_or_vault_id.number,
-            :cvv => credit_card_or_vault_id.verification_value,
-            :expiration_month => credit_card_or_vault_id.month.to_s.rjust(2, "0"),
-            :expiration_year => credit_card_or_vault_id.year.to_s
-          }
+          if credit_card_or_vault_id.is_a?(NetworkTokenizationCreditCard)
```

Comment:
Ideally we could do something like ``payment_method.tokenized?`` but this breaks when you have a string or a hash at the payment method (for example stripe token). I think we're stuck with the ``is_a?`` for the time being, unless you have other suggestions?

Braintree is the only one I know of that specifically calls it apple pay, the rest I've seen are generic. I think even the braintree solution would work for other non-apple solutions if we had one to try. I'm sure one will come along soon anyway ;)